ilabs/client/ilabs_datavault_predictor.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import absolute_import, unicode_literals
+
+from ilabs.client import ilabs_api, ilabs_datavault_api
+import time
+import json
+import logging
+import uuid
+
+def noop(*av, **kav): pass
+
+
+class ILabsDatavaultPredictor(ilabs_api.ILabsApi):
+
+    @classmethod
+    def init(cls, domain, collection, input_facet='input', output_facet='output', user_key=None, datavault_key=None, timeout=None, user_agent=None):
```

Comment:
One difference from my approach is, here you are initializing collection and IO facets as instance variables, where I am passing them as arguments in each call. This looks more efficient imo. 
 Otherwise, most of the code is very similar.